doc/tutorials/enable_ivshmem.rst

```diff
@@ -0,0 +1,151 @@
+.. _enable_ivshmem:
+
+Enable Inter-VM Communication Based on Ivshmem
+##############################################
+
+Users can use inter-vm communication based on ``ivshmem`` dm-land solution or
+hv-land solution according to the usage scenario and both solutions can be
+used at the same time. The ivshmem device using dm-land cannot communicate
+with the ivshmem device using hv-land. For more details about ivshmem inter-vm
+communication, please refer to :ref:`Ivshmem Inter-VM Communication High Level
+Design <ivshmem-hld>`
+
+Ivshmem Dm-land Usage
+*********************
+add this line as an ``acrn-dm`` boot parameter::
+
+     -s slot,ivshmem,shm_name,shm_size
+
+where
+
+-  ``-s slot``  - Specify the virtual PCI slot number
+
+-  ``ivshmem``  - Virtual PCI device name
+
+-  ``shm_name`` - Specify a shared memory name. Post-launched VMs with the same
+   ``shm_name`` share a shared memory region. The ``shm_name`` needs to start
+   with ``sos:/`` prefix. For example, ``sos:/test``
+
+-  ``shm_size`` - Specify a shared memory size. The two communicating VMs must
+   define the same size.
+
+.. note:: This device can be used with Real-Time VM (RTVM) as well.
+
+Ivshmem Hv-land Usage
+*********************
+The ``ivshmem`` hv-land is not enabled by default, users need to enable
+``ivshmem`` hv-land through ACRN XML configuration(:ref:`ACRN
+Configuration Tool <acrn_configuration_tool>`) before using it.
+
+- Enable ``ivshmem`` hv-land in ACRN XML configuration file. For example, the
+  hybrid_rt scenario and whl-ipc-i5 board XML configuration file is located in
+  ``$PWD/misc/vm_configs/xmls/config-xmls/whl-ipc-i5/hybrid_rt.xml``
+
+   - Configure ``IVSHMEM_ENABLED`` to ``y`` in ACRN scenario XML configuration
+     to enable ``ivshmem`` hv-land
+
+   - Configure ``IVSHMEM_REGION`` to specify the shared memory name, size and
+     communication VMs in ACRN scenario XML configuration. The ``IVSHMEM_REGION``
+     format such as ``shm_name,shm_size,VM IDs``:
+
+	-  ``shm_name`` - Specify a shared memory name. The name needs to start
+	   with the ``hv:/`` prefix. For example, ``hv:/shm_region_0``
+
+	-  ``shm_size`` - Specify a shared memory size
+
+	-  ``VM IDs``   - Specify the VM IDs to use the same shared memory
+	   communication and separate it with ``:``. For example, the
+	   communication between VM0 and VM2, it can be written as ``0:2``
+
+- Build the XML configuration, refer to :ref:`Build ACRN from Source <getting-started-building>`
+
+Inter-VM Communication Example
+******************************
+
+**The following example uses inter-vm communication(dm-land) between two
+Linux-based post-launched VMs (VM1 and VM2)**
+
+.. note:: An ``ivshmem`` Windows driver exists and can be found `here <https://github.com/virtio-win/kvm-guest-drivers-windows/tree/master/ivshmem>`_
```

Comment:
Is this an officially supported repo?  There is no licensing information listed, nor in the code headers, and there's no readme.  We should probably add some more disclaimer information here about what folks will find when they click on this link and that it's not supported by the ACRN project.

Reply:
I am not sure if it is official.

Could you give more suggestions about this or I need to remove it?

Reply:
@dbkinder , there is licensing information one level up: https://github.com/virtio-win/kvm-guest-drivers-windows. Doesn't that automatically cover all folders in the repo? If so, I think we're covered and this does look an official repo to me.